src/server/tiered_storage.cc

```diff
@@ -359,6 +359,54 @@ void TieredStorage::Free(PrimeIterator it, DbTableStats* stats) {
   stats->tiered_size -= len;
 }
 
+size_t TieredStorage::GetBinSize(size_t blob_len) {
+  CHECK(blob_len <= kMaxSmallBin);
+  unsigned bin_index = SmallToBin(blob_len);
+  return kSmallBins[bin_index];
+}
+
+void TieredStorage::Defrag(DbIndex db_index, PrimeIterator it) {
```

Comment:
Last time we spoke I suggested to iterate over the page_refcnt_ in defrug task and if you find a page under utilized load its values. Are you goint to change this PR according to this suggestion?

Reply:
I just committed the change to implement the above.